src/main/java/com/rehab/model/Cure.java

```diff
@@ -0,0 +1,21 @@
+package com.rehab.model;
+
+public class Cure extends AbstractNamedEntity {
+    private CureType cureType;
+
+    public Cure() {
```

Comment:
When you define default constructor with empty body then it is possible that getCureType returns NULL. You should either get rid of this constructor or initialize cureType with some default value.

Reply:
Initialized cureType with MEDICINE in commit c04ab95